### PR TITLE
Expanded tests

### PR DESCRIPTION
**Pull request by tokengeek**
An assortment of specs and test changes I found on old branches.

Support work for existing behaviours to support refactoring.

### CODE REVIEW

File: tests/helper.rb [hunk 1]
```diff
@@ -13,6 +13,12 @@
   FOG_TESTING_TIMEOUT = ENV['FOG_TEST_TIMEOUT'] || 2000
   Fog.timeout = 2000
   Fog::Logger.warning "Setting default fog timeout to #{Fog.timeout} seconds"
+
+  # These sets of tests do not behave nicely when running mocked tests
+  Thread.current[:tags] << '-serverlove'
```
Comment (plribeiro3000) [inline tests/helper.rb]: Is there any serverlove code left here?

Comment (tokengeek) [inline tests/helper.rb]: It looks like since I originally wrote this the tests were removed. Which means this isn't much use.

I'll update accordingly.

**Comment by plribeiro3000** [comment]
:+1: @tokengeek

**Comment by geemus** [comment]
Thanks!

**Pull request merged**